### PR TITLE
Enable RBAC support

### PR DESCRIPTION
**Pull request by sudhakar-nallam**
Added JWT Token based Auth support

### CODE REVIEW

File: plugins/rest_api_plugin.py [hunk 1]
```diff
@@ -64,6 +70,7 @@ def get_config_boolean_value(section, key, default_value):
 airflow_rest_api_plugin_http_token_header_name = get_config_string_value("rest_api_plugin", "REST_API_PLUGIN_HTTP_TOKEN_HEADER_NAME", "rest_api_plugin_http_token")
 airflow_expected_http_token = get_config_string_value("rest_api_plugin", "REST_API_PLUGIN_EXPECTED_HTTP_TOKEN", None)
 web_authentication_enabled = configuration.getboolean("webserver", "AUTHENTICATE")
+rbac_authentication_enabled = configuration.getboolean("webserver", "RBAC")
```
Comment (rssanders3) [inline plugins/rest_api_plugin.py:73]: Set default in the event that RBAC is not included.

File: plugins/rest_api_plugin.py [hunk 2]
```diff
@@ -296,7 +303,8 @@ def get_config_boolean_value(section, key, default_value):
             {"name": "ignore_first_depends_on_past", "description": "Ignores depends_on_past dependencies for the first set of tasks only (subsequent executions in the backfill DO respect depends_on_past).", "form_input_type": "checkbox", "required": False},
             {"name": "subdir", "description": "File location or directory from which to look for the dag", "form_input_type": "text", "required": False},
             {"name": "pool", "description": "Resource pool to use", "form_input_type": "text", "required": False},
-            {"name": "dry_run", "description": "Perform a dry run", "form_input_type": "checkbox", "required": False}
+            {"name": "dry_run", "description": "Perform a dry run", "form_input_type": "checkbox", "required": False},
+            {"name": "reset_dagruns", "description": "if set, the backfill will delete existing backfill-related DAG runs and start anew with fresh, running DAG runs", "form_input_type": "checkbox", "required": False}
```
Comment (rssanders3) [inline plugins/rest_api_plugin.py]: Were we able to get this to work? I recall you saying you had issues with it prompting you to confirm yes or no

File: plugins/rest_api_plugin.py [hunk 3]
```diff
@@ -564,37 +582,74 @@ def get_dagbag():
     def get_argument(request, arg):
         return request.args.get(arg) or request.form.get(arg)
 
+    # overrides BaseView method to show/hide the menu links dynamically
+    def is_visible(self):
+        if web_authentication_enabled and current_user.is_authenticated == False:
```
Comment (rssanders3) [inline plugins/rest_api_plugin.py:586]: Would `current_user.is_authenticated` fail if authentication was not enabled?

**Review by rssanders3 — CHANGES_REQUESTED** [review]

File: plugins/rest_api_plugin.py (hunk 2 above)
Comment (sudhakar-nallam) [reply plugins/rest_api_plugin.py]: This is not working if there is any failed instances. So, I have reverted this change.

File: plugins/rest_api_plugin.py (hunk 1 above)
Comment (sudhakar-nallam) [reply plugins/rest_api_plugin.py:73]: It is returning false in case RBAC is not set. Verified in both Python 2.x and 3.x

**Review by rssanders3 — APPROVED** [review]

**Pull request merged**